acceptance/tests/resource/package/common_package_name_in_different_providers.rb

```diff
@@ -30,26 +30,35 @@
   end
 end
 
-def verify_state(hosts, pkg, state, match)
-  hosts.each do |agent|
-    cmd = rpm_provider(agent)
-    # Note yum lists packages as <name>.<arch>
-    on agent, "#{cmd} list installed" do
-      method(match).call(/^#{pkg}\./, stdout)
-    end
+def verify_state(host, pkg, state, match)
+  cmd = rpm_provider(host)
+  # Note yum lists packages as <name>.<arch>
+  on host, "#{cmd} list installed" do
+    method(match).call(/^#{pkg}\./, stdout)
+  end
 
-    on agent, 'gem list --local' do
-      method(match).call(/^#{pkg} /, stdout)
-    end
+  on host, "gem list --local", :environment => {'PATH' => agent_path(host)} do
+    method(match).call(/^#{pkg} /, stdout)
   end
 end
 
-def verify_present(hosts, pkg)
-  verify_state(hosts, pkg, '(?!purged|absent)[^\']+', :assert_match)
+def verify_present(host, pkg)
+  verify_state(host, pkg, '(?!purged|absent)[^\']+', :assert_match)
 end
 
-def verify_absent(hosts, pkg)
-  verify_state(hosts, pkg, '(?:purged|absent)', :assert_no_match)
+def verify_absent(host, pkg)
+  verify_state(host, pkg, '(?:purged|absent)', :assert_no_match)
+end
+
+def agent_path(host)
+  # This is necessary so that the puppet-agent gem binary is in the path
+  # during `puppet apply` runs. Otherwise, the gem provider is considered
+  # unsuitable.
+  if @options[:type] == 'aio'
+    "#{host['privatebindir']}:${PATH}"
+  else
+    '${PATH}'
+  end
```

Comment:
If we want to make sure we use puppet's gem command so we can install a gem in a manifest, we should just use the `puppet_gem` provider, e.g.:

```
package {'gem-guid':
  provider => puppet_gem,
  name => 'guid',
  ensure => installed,
}
```

That way we don't have to monkey with pathing.